pkg/k8shandler/exec.go

```diff
@@ -0,0 +1,81 @@
+package k8shandler
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/operator-framework/operator-sdk/pkg/k8sclient"
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/remotecommand"
+)
+
+type ExecConfig struct {
+	namespace      string
+	labelName      string
+	labelPrefix    string
+	containerName  string
+	command        []string
+	kubeConfigPath string
+	masterURL      string
+}
+
+func PodExec(config *ExecConfig) (*bytes.Buffer, *bytes.Buffer, error) {
+	var (
+		execOut bytes.Buffer
+		execErr bytes.Buffer
+		esPod   *v1.Pod
+	)
+
+	client := k8sclient.GetKubeClient()
+
+	pods, err := client.CoreV1().Pods(config.namespace).List(metav1.ListOptions{})
+	if err != nil {
+		return new(bytes.Buffer), new(bytes.Buffer), fmt.Errorf("couldn't get pods in namepaces %s. Error: %v", config.namespace, err)
+	}
+	for _, pod := range pods.Items {
+		labels := pod.GetLabels()
+		if labels[config.labelName] != "" && strings.HasPrefix(labels[config.labelName], config.labelPrefix) {
+			esPod = &pod
+			break
+		}
+	}
+	if esPod == nil {
+		return nil, nil, fmt.Errorf("couldn't find pod in namespace %s with labesl %s=prefix(%s)", config.namespace, config.labelName, config.labelPrefix)
+	}
+
+	execRequest := client.CoreV1().RESTClient().Post().
+		Resource("pods").
+		Name((*esPod).Name).
+		Namespace((*esPod).Namespace).
+		SubResource("exec")
+
+	execRequest.VersionedParams(&v1.PodExecOptions{
+		Container: config.containerName,
+		Command:   config.command,
+		Stdout:    true,
+		Stderr:    true,
```

Comment:
@ewolinetz we don't need it now, but maybe it's a good idea to make stdout, stdin and tty configurable.